modules/kotest/src/main/kotlin/org/testcontainers/containers/ContainerPerTestListener.kt

```diff
@@ -0,0 +1,44 @@
+package org.testcontainers.containers
+
+import io.kotest.core.listeners.TestListener
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+
+
+/**
+ * [ContainerPerTestListener] starts the given [genericContainers] before execution of each test in the spec
+ * and stops after execution of each test.
+ *
+ * [genericContainers] can any of [GenericContainer] [ContainerisedDockerCompose] [LocalStackContainer] etc.
+ *
+ * This should be use when you want fresh container for each test.
+ *
+ * @see[ContainerPerSpecListener]
+ * Usage:
+ *
+ * class RedisRepositoryTest: StringSpec() {
+ *    private val redisContainer = GenericContainer<Nothing>("redis")
+ *
+ *    override fun listeners(): List<TestListener> {
+ *       return super.listeners() + ContainerPerTestListener(redisContainer)
```

Comment:
Not sure about the Kotlin idiomatic way here, but isn't there something like an annotation we could use? Like we do in the other test frameworks?

Reply:
We use annotations in ```AnotationSpec``` style in kotest, but that style is only to reduce the learning curve to people coming from jUnit or other similar background, as far as I know kotest extension points are used in this style.

Reply:
Annotations are not idiomatic in frameworks that take advantage of the DSL capabilities provided by languages like Kotlin. Why use an annotation when you can use a function and all the power that comes with functions. 

That being said, the example in the documentation has too much boilerplate. It can be simplified to use one additional line, like with annotations:

```kotlin
 class RedisRepositoryTest: FunSpec({

    // create the TestContainer
    val redisContainer = GenericContainer<Nothing>("redis")

    // register it with kotest to be started / stopped with the tests
    listener(ContainerPerSpecListener(redisContainer))

    test("should be able to connect redis server") {
      // your test goes here
   }
})
```

Anyone, if they wish, can compose the ContainerPerSpecListener and add extra capabilities. You can't do that as easily with annotations.

Reply:
Sure @sksamuel

Reply:
Done